pkg/provider/acceptance_type_test.go

```diff
@@ -0,0 +1,101 @@
+package provider
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+
+	"github.com/MaterializeInc/terraform-provider-materialize/pkg/materialize"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/jmoiron/sqlx"
+)
+
+func TestAccType_basic(t *testing.T) {
+	typeName := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTypeResource(typeName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTypeExists("materialize_type.test"),
+					resource.TestCheckResourceAttr("materialize_type.test", "name", typeName),
```

Comment:
There are a few other attributes we should track like the `schema_name`, `database_name`, `qualified_sql_name`, `list_properties`, `map_properties`, `category`